src/output.c

```diff
@@ -290,19 +288,15 @@ output_config_apply(struct server *server,
 		}
 
 		/* Only do Layout specific actions if the commit went trough */
-		if (need_to_add) {
-			wlr_output_layout_add_auto(server->output_layout, o);
+		if (output_enabled) {
+			/* moves the output in the layout if already added */
+			wlr_output_layout_add(server->output_layout, o,
+				head->state.x, head->state.y);
```

Comment:
This will unconditionally disable the automatic layout. Instead, we should compare the current position within the layout with `head->state.x/y` and if they are the same just do nothing. This is an pre-existing issue with the current master branch as well, that one unconditionally uses `wlr_output_layout_move()` which does the same.

In terms of keeping this PR as small as possible I'd suggest reverting all changes in `src/output.c` and instead just rename `wlr_output_layout_move()` to `wlr_output_layout_add()` (which btw is a pretty strange name / double-use API).